lib/ThreeScaleClient.php

```diff
@@ -95,7 +110,18 @@ public function getHost() {
    */
   public function authorize($appId, $appKey = null, $serviceId = null, $usage = null) {
     $url = "http://" . $this->getHost() . "/transactions/authorize.xml";
-    $params = array('provider_key' => $this->getProviderKey(), 'app_id' => $appId);
+    $params = array('app_id' => $appId);
+
+    $providerKey = $this->getProviderKey();
+    $serviceToken = $this->getServiceToken();
```

Comment:
I was trying Eric's way on java plugin, but function overloading is not supported in php and thus followed this approach.

Let me check how ruby is implemented.